Library/Homebrew/test/cmd/--env_spec.rb

```diff
@@ -1,43 +1,38 @@
 describe "brew --env", :integration_test do
   it "prints the Homebrew build environment variables" do
     expect { brew "--env" }
-      .to output(/CMAKE_PREFIX_PATH="#{Regexp.escape(HOMEBREW_PREFIX)}[:"]/).to_stdout
```

Comment:
I don't see CC being set either from `brew --env` or `brew install --env=super --interactive hello` then doing `brew --env`. Here are the options that are available to me, AFAICT:

```
HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
MAKEFLAGS: -j4
CMAKE_INCLUDE_PATH: /usr/include/libxml2:/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.13
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/bin:/bin:/usr/sbin:/sbin
```

So the best ones to verify might be:
- `HOMEBREW_CC` matches `clang`
- `HOMEBREW_CXX` matches `clang++`

and then we could also check that `PATH` contains each of:
- `/usr/bin`
- `/bin`
- `/usr/sbin`
- `/sbin`

What do you think @MikeMcQuaid? I think I might be capable of implementing that with my limited ruby. Also please let me know if I should squash everything, or use multiple commits.

Not so sure why `CMAKE_LIBRARY_PATH` and `CMAKE_INCLUDE_PATH` are getting set, but it's not harming me, so I'll leave that alone.

Reply:
`HOMEBREW_CC: clang`, sorry. It's sufficient to check that instead of `CMAKE_PREFIX_PATH`.

> and then we could also check that PATH contains each of:

Don't think we need to do that.

Reply:
OK, so no expansion of the env tests beyond `HOMEBREW_CC=clang`? Could be good to test `ACLOCAL_PATH` has content since the env tests are fairly sparse, but I'm happy leaving it alone too.